shared-libs/cht-datasource/src/person.ts

```diff
@@ -87,6 +87,35 @@ export namespace v1 {
       return curriedFn;
     };
 
+  const getPeopleGenerator = () => (context: DataContext) => {
+    assertDataContext(context);
+
+    /**
+     * Creates an iterator-like object for fetching batches of people data.
+     * @param personType - The type of person to fetch
+     * @throws throws an error if the provided personType is invalid.
+     * @returns An iterator-like object with a next method for fetching data.
+     */
+    const curriedFn = async function* (personType: ContactTypeQualifier): AsyncGenerator<Person[], void> {
+      assertTypeQualifier(personType);
+      const limit = 100;
+      let skip = 0;
+
+      while (true) {
```

Comment:
ESLint complains about an unnecessary conditional. As much as I like ESLint, I think this is one of the few exceptions where we silence it. What do you think @jkuester @sugat009 ?
```suggestion
      // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
      while (true) {
```

Reply:
My vote is for the venerable do-while:

```ts
    const curriedFn = async function* (personType: ContactTypeQualifier): AsyncGenerator<Person, void> {
  const getPage = context.bind(v1.getPage);
  const limit = 100;
  let skip = 0;
  const hasMoreResults = () => skip && skip % limit === 0;
  do {
    const docs = await getPage(personType, limit, skip);
    for (const person of docs) {
      yield person;
    }

    skip += docs.length;
  } while (hasMoreResults());
};
return curriedFn;
};
```

Note that I also think we need to change from returning an array of persons (yielding each batch) to just yielding one person at a time.  I think that is going to be the cleanest for code trying to consume this (otherwise the consuming code ends up in these wierd nested loops).

Reply:
Also, this generator logic is pretty generic.  I guess we could probably re-use it for the `Place.v1.getAll` if you allow the actual `getPage` function to be injected.  If we go that round, I guess we could put the shared function in `libs/data-context.ts`.

Reply:
I did a performance test(nothing serious just timed the entry and exit where the function `getAll` is called and looped until the end) for sanity for the current implementation of returning arrays of people VS retuning singular people for data with 27k docs and almost 10k people and the performance is pretty much the same. So, I think if consumers could avoid the nested for loop, we should implement @jkuester's suggestion from above.